### ISSUE TITLE
Any reason Style/Dir shouldn't detect File.dirname(__FILE__) ?

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#8770]
## Expected behavior

I'm trying to understand if there is a reason *not* to expand the Style/Dir cop to also detect simpler forms of `__FILE__` that can be replaced with `__dir__`. It seems like `File.dirname(__FILE__)` can / should also be converted to `__dir__` in addition to `File.expand_path(File.dirname(__FILE__))`

## Actual behavior

Right now only using `__FILE__` to determine the absolute path of a directory is detected.

## Steps to reproduce the problem

scan this:
```ruby
File.expand_path(File.dirname(__FILE__))

File.dirname(__FILE__)
```

## RuboCop version

```
master
```

**Comment by marcandre** [comment]
The reason is that `__dir__` is always absolute, while `__FILE__` is relative, so you may get `File.dirname(__FILE__)` as `"."` and `__dir__` as `"/Users/mal/temp"`. This may or may not give different results.

If someone is trying to copy the current code to somewhere else, s/he could do `path = "/tmp/absolute/#{File.dirname(__FILE__)}"` and the cop would break this.

That being said, we could make an allowlist for simpler cases where it is equivalent:

```ruby
File.write("#{File.dirname(__FILE__)}<anything>", <anything>")
YAML.load("#{File.dirname(__FILE__)}<anything>")
# etc
```

**Comment by tas50** [comment]
```ruby
File.expand_path(File.dirname(__FILE__))
```
Should be safe though since you're always expanding '.' when it comes down to it. I see that pattern used a lot.

**Comment by tas50** [comment]
Thanks for the extra info @marcandre
